components/server/src/workspace/workspace-service.ts

```diff
@@ -222,6 +223,31 @@ export class WorkspaceService {
         return filtered;
     }
 
+    async listWorkspaceSessions(
+        userId: string,
+        organizationId: string,
+        from: Date,
+        to: Date,
+        limit: number,
+        offset: number,
+    ): Promise<WorkspaceSession[]> {
+        await this.auth.checkPermissionOnOrganization(userId, "read_sessions", organizationId);
+
+        // check from is before to
+        if (from >= to) {
+            throw new ApplicationError(ErrorCodes.INVALID_VALUE, "from must be before to");
```

Comment:
Better to use `ErrorCodes.BAD_REQUEST` because we have converter for it

https://github.com/gitpod-io/gitpod/blob/se/workspace-sessions/components/public-api/typescript-common/src/public-api-converter.ts#L591-L593